src/01_github-wrangling/06_commits_nmrc_jbsc.sql

```diff
@@ -0,0 +1,84 @@
+create materialized view gh.commits_freq as (
+    with a as (
+        SELECT committed_date,
+            login,
+            additions,
+            deletions,
+            count(*)
+        FROM gh.commits_dd
+        group by committed_date,
+            login,
+            additions,
+            deletions
+    )
+    select *
+    from a
+    where count > 1
+    order by count desc,
+        committed_date,
+        login,
+        additions,
+        deletions
+);
+create materialized view gh.commits_freq_0 as (
+ SELECT a.slug,
+    a.committed_date,
+    a.login,
+    a.additions,
+    a.deletions
+   FROM gh.commits_dd a
+   JOIN gh.commits_freq b
+   ON a.committed_date = b.committed_date
+   AND a.login = b.login
+   AND a.additions = b.additions
+   AND a.deletions = b.deletions
+);
+create materialized view gh.commits_freq_1 as (
+    with a as (
+        SELECT slug,
+            max(committed_date),
+            min(committed_date)
+        FROM gh.commits_freq_0
+        group by slug
+    ),
+    b as (
+        select distinct on (slug) slug,
+            max - min as duration
+        from a
+        order by slug,
+            duration desc
+    )
+    select distinct a.*
+    from gh.commits_freq_0 a
+    join b
+    on a.slug = b.slug
+);
+create materialized view gh.commits_dd_nmrc_jbsc as (
```

Comment:
-- this one does a union all on the commits_dd (with all classified as to_keep = false) and the table flattened duration table (to_keep = true). my understanding is that the union all works like bind_rows in tidyverse so that bayoan is essentially adding in more rows to the commits_dd table and then he sorts on the to_keep variable so that the distinct only takes the first row of all the duplicates. this means that we end up with only one of the multi-repo commits bc we are taking the one commit from all the >1 repos. so the end product is a table that has all of the no multi-repo commits + the longest duration multi-repo commits, resulting in a table of 312,955,618 rows that need to run through cost calculations next.